raiden-ts/src/transfers/reducer.ts

```diff
@@ -202,52 +187,33 @@ function transferExpireSuccessReducer(
   return state;
 }
 
-function transferUnlockProcessedReducer(
+function transferStateReducer(
   state: RaidenState,
-  action: transferUnlockProcessed,
+  action: transferProcessed | transferUnlockProcessed | transferExpireProcessed | transferRefunded,
 ): RaidenState {
   const secrethash = action.meta.secrethash;
   if (!(secrethash in state.sent)) return state;
-  return {
-    ...state,
-    sent: {
-      ...state.sent,
-      [secrethash]: {
-        ...state.sent[secrethash],
-        unlockProcessed: timed(action.payload.message),
-      },
-    },
-  };
-}
 
-function transferExpireProcessedReducer(
-  state: RaidenState,
-  action: transferExpireProcessed,
-): RaidenState {
-  const secrethash = action.meta.secrethash;
-  if (!(secrethash in state.sent)) return state;
-  return {
-    ...state,
-    sent: {
-      ...state.sent,
-      [secrethash]: {
-        ...state.sent[secrethash],
-        lockExpiredProcessed: timed(action.payload.message),
-      },
-    },
-  };
-}
+  let key: 'transferProcessed' | 'unlockProcessed' | 'lockExpiredProcessed' | 'refund';
+  if (transferProcessed.is(action)) {
+    key = 'transferProcessed';
+  } else if (transferUnlockProcessed.is(action)) {
+    key = 'unlockProcessed';
+  } else if (transferExpireProcessed.is(action)) {
+    key = 'lockExpiredProcessed';
+  } else if (transferRefunded.is(action)) {
+    key = 'refund';
+  } else {
+    return state;
```

Comment:
You don't need the `else` if you check all possibilities.

Reply:
The problem is that for `switch` to work I have to work with the `action.type` instead of the `is(action)`. The const/ternary approach will be more confusing due to the multiple chained ternary operators.

Finally, I am aware that else is not logically needed, but without the `else` there `tsc` doesn't consider the checks exhaustive so it complains about using the variable before assigning any value to it.

Reply:
If you wanna stick with if/else, you can safely replace the last `else if` with just `else`, so to make it exhaustive. But this was just nitpicking, don't mind it. Thank you =)